### ISSUE TITLE
Problems with spell check Macintosh 

### ISSUE THREAD

**Issue by rll307** [rstudio/rstudio#7018]
Trying to use Spell Check under MacOs Catalina Rstudio Version 1.3.959

The following error pops up:

check spelling An error has occurred: (TypeError) : Cannot read property 'check' of undefined

The erros happens whichever language I choose for checking

**Comment by jmcphers** [comment]
Sorry, we can't reproduce this without some additional details. Can you share any of the documents which trigger this error for you? Which languages are you choosing for checking? Have you installed any custom dictionaries?

**Comment by rll307** [comment]
Hi, 

I tried in Brazilian Portuguese and English. Here is a copy go the document. 
[01.Rmd.zip](https://github.com/rstudio/rstudio/files/4713764/01.Rmd.zip)

**Comment by jmcphers** [comment]
@adamconroy can you reproduce?

**Comment by adamconroy** [comment]
I can try in a little bit but my guess off the bat here is it's just not loading the dictionaries. Possibly this was an upgrade from a previous version and the dictionaries aren't where it thinks they should be. @rll307 if you open up the network tab (right click -> inspect -> network) and then right click reload/change dictionaries do you see any errors? Also do you see any errors in the console tab?

**Comment by stibu81** [comment]
I have the same problem on Ubuntu after I updated RStudio-Desktop. I uninstalled RStudio, deleted all the configuration files following [this article](https://support.rstudio.com/hc/en-us/articles/200534577-Resetting-RStudio-Desktop-s-State) and reinstalled RStudio. After that, spell check in English worked again.

However, I need spell checking in German. So I went to Global Options - Spelling and clicked "Install More Languages..." and the problems start again. Unfortunately, there is a lot of indeterminism in this. I have tried various things and depending on what exactly I do, the error message reoccurs immediately or only after I closed RStudio. Each time, I can get the spell checker working by deleting .config/rstudio (where it seems the dictionaries are located), but never for long.

The error appears for any file, also the Example-File that is generated, when I create a new Rmd-File.

If I open up the Network tab, as suggested, and press F7, the error message shows also in that window as "Class$S314: (TypeError) : Cannot read property 'check' of undefined    rstudio-0.js:6024". When I pick another language or choose "Update Dictionaries...", I see no error

This is a serious problem for me, since I write a lot of reports and slides in RStudio and really need the spell checker. So I am willing to invest some time in order to help. What would be useful for you? I could, for instance, describe some of the scenarios that I tried in some detail, if this helps.

**Comment by rll307** [comment]
Hi

I tried something similar to @stibu81 . But it did not work out for me as well. I need spell check in both English and Portuguese. I am using brew as a package manager. 

I tried to  install a personalised dictionary, yet no success.

My solution was downgrading. After that I could get spelling check all right.

**Comment by adamconroy** [comment]
@rll307 this should be fixed in the 1.3 patch release. Sorry about that. Can you confirm that you're trying to run with realtime spellcheck turned off (In Global Options -> Spelling)? If you turn it on does your problem go away? If that isn't the case then the PR I just made won't fix it and I'll need some more information.

**Comment by rll307** [comment]
Hi. @adamconroy 

Here is my system configuration:

- *R* version 3.6.3 (2020-02-29) -- Holding the Windsock
- *Rstudio* Version 1.3.959
- *System* MacOS Catalina

The problem seemed to go away when you have only English as a language for spells checking. The problem persists when I try to check the spelling in Portuguese. The system actually tells me that real time spellchecking with Portuguese in unavailable.

I attached some screenshots. I hope they might help:

![Screenshot 2020-06-09 at 10 13 47](https://user-images.githubusercontent.com/14953924/84152962-b2561600-aa3b-11ea-8d55-3a4b803c471a.png)
![Screenshot 2020-06-09 at 10 14 07](https://user-images.githubusercontent.com/14953924/84152974-b6823380-aa3b-11ea-99c1-6d252e4f214d.png)
![Screenshot 2020-06-09 at 10 15 34](https://user-images.githubusercontent.com/14953924/84152993-ba15ba80-aa3b-11ea-9ed2-698e979001f6.png)
![Screenshot 2020-06-09 at 10 15 59](https://user-images.githubusercontent.com/14953924/84153001-bd10ab00-aa3b-11ea-8fcd-6e650ec23129.png)
![Screenshot 2020-06-09 at 10 16 20](https://user-images.githubusercontent.com/14953924/84153020-c0a43200-aa3b-11ea-96ef-b64d8a97ee67.png)

**Comment by adamconroy** [comment]
@rll307 ok this is the problem that I just fixed. The fix will be in the next 1.3 patch. Sorry for the difficulties and thank you for reporting it.

**Comment by jmcphers** [comment]
@rll307 Here's the fixed build -- can you confirm it doesn't reproduce there?

https://rstudio.com/products/rstudio/download/preview/

**Comment by stibu81** [comment]
@adamconroy @jmcphers Thanks for the quick fix. I can confirm that spell check in German and English now works under Ubuntu 18.04.

**Comment by jmcphers** [comment]
Great, glad to hear it!

**Comment by rll307** [comment]
Hi,

I performed some tests and it did not worked for me.

1. I uninstalled Rstudio completely according to this [article](https://support.rstudio.com/hc/en-us/articles/200534577-Resetting-RStudio-Desktop-s-State)
2. I installed the latest version of Rstudio in this [repository](https://rstudio.com/products/rstudio/download/preview/)
3. I tried to spellcheck in English than in Portuguese (Brazilian).

Here is the outcome:

1. Spellchecking in English worked just fine
2. Spellchecking in Portuguese returned me the same error

Then I restarted Rstudio and tried again, it started spellchecking in Portuguese using an English dictionary.

Bellow are some pics to illustrate the process. To able to carry on my research (I am a University professor)  I am downgrading to RStudio-1.2.5042. 

![Screenshot 2020-06-10 at 08 20 39](https://user-images.githubusercontent.com/14953924/84263772-7764e880-aaf6-11ea-94f8-74d49f206f80.png)
![Screenshot 2020-06-10 at 08 21 11](https://user-images.githubusercontent.com/14953924/84263788-7fbd2380-aaf6-11ea-8d7c-0e4a98a20a75.png)
![Screenshot 2020-06-10 at 08 21 21](https://user-images.githubusercontent.com/14953924/84263798-8481d780-aaf6-11ea-87e3-84cfeb37f065.png)
![Screenshot 2020-06-10 at 08 21 43](https://user-images.githubusercontent.com/14953924/84263825-8ea3d600-aaf6-11ea-87d6-d625917e5264.png)
![Screenshot 2020-06-10 at 08 22 38](https://user-images.githubusercontent.com/14953924/84263835-9499b700-aaf6-11ea-8ef7-f65c479b73ea.png)
![Screenshot 2020-06-10 at 08 24 13](https://user-images.githubusercontent.com/14953924/84263843-98c5d480-aaf6-11ea-9cac-0be3867e38f5.png)

**Comment by francarotti** [comment]
```
RStudio version: 1.3.1023-1 
OS: Windows 10 Home Desktop (Version 2004)
R Version: 4.0.0
```

Posting to extend on the issue, this error also happens on Windows 10 when I try to spell check (F7) in Brazilian Portuguese. Downgrading to 1.2.5042 fixed this issue (thanks @rll307 !). 

I'll be happy to test another build, but will be available only on Friday.

**Comment by adamconroy** [comment]
@rll307 @francarotti sorry again for the issues, the latest daily of v1.3 should work for you now. Let me know if it doesn't!

**Comment by rll307** [comment]
@adamconroy 

I just ran a test on Brazilian Portuguese and it worked partially. 

# My info
- Mac OS: 10.15.5 (19F101)
- R version 4.0.1 (2020-06-06) -- "See Things Now"
- RStudio Version 1.3.1031

# My procedures

1. I created a new rmarkdown file and changed the spelling to  Brazilian Portuguese
1. I typed a series of randon words (one with a typo)
1. I ran a ordinay spellcheck

# My results
1. There was no crash
1. Words spelled correctly were all right
1. Words with typo were replaced for an empty space

Actually, the spellcheck window does not show any information regarding Portuguese. Here are some screenshots.

Hope it helps to sort it out

![Screenshot 2020-06-17 at 21 50 45](https://user-images.githubusercontent.com/14953924/84965605-ba6e1f80-b0e5-11ea-8df8-96c1661455df.png)
![Screenshot 2020-06-17 at 21 51 10](https://user-images.githubusercontent.com/14953924/84965607-bcd07980-b0e5-11ea-9b75-f2b54491e3a9.png)

**Comment by adamconroy** [comment]
@rll307 I just tested locally and suggestions are showing up for...some words but not the same as 1.2. Something has been lost in the changes, sorry about that. I have a 1.2 build on my laptop now that I will test against and fix this tomorrow.

**Comment by jmcphers** [comment]
Should be fixed in 1.3.1039.

**Comment by francarotti** [comment]
```
RStudio version: 1.3.1039-1
OS: Windows 10 Home Desktop (Version 2004)
R Version: 4.0.0
```

Thanks for working on this issue. The error doesn't appear anymore words with typos are correctly corrected. 

Unfortunately, spell checking is ignoring some words that should have accents, like Nao (should be Não), idolo (ídolo), cafe (café). For these cases, suggestions doesn't match the correct word. For similar words like pao (pão), spell checker works fine.

**Comment by R-Hannibal** [comment]
@francarotti thanks so much for following up on this issue.

In the examples that you gave (Nao should be Não, idolo should be ídolo, etc.), are you seeing a difference in functionality between 1.2 and 1.3 releases (in other words, does it work correctly in 1.2 builds)?

**Comment by francarotti** [comment]
@R-Hannibal Sorry, I didn't tested these examples in 1.2 before posting. There is no difference in functionality, both versions behave in the same way ignoring words that should be accentuated.

Because this was an issue related with the spell checker throwing an error, the problem was solved and could be closed again. 

What I was addressing is an issue with the dictionary used by default in RStudio. I would like to suggest the dictionary used by LibreOffice that can be found [here](https://pt-br.libreoffice.org/projetos/vero/) in the "spell checker" label. Unfortunately I couldn't find an english version for this page. I installed both .dic and .aff files and the examples that I gave and some other cases were correctly addressed.  

Since this is an unrelated issue, I can post this properly as an enhancement/another issue. Thanks!

**Comment by R-Hannibal** [comment]
Seriously, @francarotti thank YOU so much for helping us to improve. We are very grateful to have awesome users like yourself.

Your assessment is exactly correct and we will close this issue and look to address the enhancement in a future release (if you don't mind mentioning [this issue number](https://github.com/rstudio/rstudio/issues/7018) in that new issue it will help us to connect the dots). Thanks again.
